spec/OneClickInstall_spec.rb

```diff
@@ -44,20 +54,195 @@
+      it "converts map structure (eg. @repositories) to a list of maps with a key element" do
+        flattened = subject.makeXMLFriendly(software)
+        expect(flattened).to match_array([
+          {"recommended"=>"true", "action"=>"remove", "type"=>"pattern", "key"=>"sap_server"},
+          {"recommended"=>"false", "action"=>"install", "type"=>"package", "key"=>"tftp"},
+          {"recommended"=>"true", "action"=>"install", "type"=>"package", "key"=>"vim"},
+          {"recommended"=>"true", "action"=>"install", "type"=>"pattern", "key"=>"x11"}])
+      end
+    end
+  end
+
+  describe "#fromXMLFriendly" do
+    context "convert list of maps to map" do
+      it "converts back from makeXMLFriendly to original structure" do
+        flattened = [
+          {"recommended"=>"true", "action"=>"remove", "type"=>"pattern", "key"=>"sap_server"},
+          {"recommended"=>"false", "action"=>"install", "type"=>"package", "key"=>"tftp"},
+          {"recommended"=>"true", "action"=>"install", "type"=>"package", "key"=>"vim"},
+          {"recommended"=>"true", "action"=>"install", "type"=>"pattern", "key"=>"x11"}]
+        unflattened = subject.fromXMLFriendly(flattened)
+        expect(unflattened).to match_array(software)
+      end
+    end
+  end
+
+  describe "#ToXML" do
+    context "serialize OneClickInstall data structure" do
+      it "serializes to XML file, according to OneClickInstall data structure" do
+        filename = File.join(DATA_PATH, "serialized_result.xml")
+        subject.ToXML(filename)
+        expected = File.join(DATA_PATH, "serialized_expected.xml") 
+        expect(IO.read(filename)).to eql(IO.read(expected))
+        File.delete(filename)
+      end
+    end
+  end
+
+  describe "#FromXML" do
+    context "deserialize OneClickInstall data structure" do
+      it "deserializes according to OneClickInstall data structure to XML" do
+        filename = File.join(DATA_PATH, "to_deserialize.xml")
+        subject.FromXML(filename) 
+        expect(subject.instance_variable_get(:@software)).to match_array({"zdoom"=>{"action"=>"install", "recommended"=>"false", "type"=>"package"}})
+        expect(subject.instance_variable_get(:@repositories)).to match_array({"https://ftp.gwdg.de/pub/linux/misc/packman/suse/openSUSE_Tumbleweed/"=>
+  {"name"=>"packman", "recommended"=>"false"}})
+        expect(subject.instance_variable_get(:@remainSubscribed)).to be true
+        expect(subject.instance_variable_get(:@summary)).to be_empty
+        expect(subject.instance_variable_get(:@description)).to be_empty
+        expect(subject.instance_variable_get(:@name)).to be_empty
+      end
+    end
+  end
+
+  describe "#Load" do
+    context "load YMP file" do
+      it "loads the Metapackage data from the YML file supplied for further processing" do
+        subject.Load(filepath)
+        expect(subject.instance_variable_get(:@repositories)[urls[0]]).to match_array("name"=>"openSUSE:Factory",
```

Comment:
checking other unit tests and given that file is fixed, we could use regex here to not have so much code hardcoded, just and idea, YaST developers might have some guidelines for that, perhaps is less error-prone (just thinking loud)

Reply:
I was thinking that it would be adequate if instead of compairing the whole array content, to use something like include("part of the expected content"). But what do you mean be regex? Can you give an example?

Reply:
yes, the regex for ruby, array responds to `#include?` so I guess something like: `.to include(/regex1/, /regex2/)` if that syntax is correct that I'm not sure, the problem with this, is that is not exact match. So I guess we would need to match each field with an `expect` (summary, description, etc.) and a regex for the value, that would make it lighter.

Reply:
I did some experimenting and if it looks like if I use include, I have to either give the exact value of the key or not check for the key at all (or use some weird syntax with extra functions, I'm still not sure how to do that). So, I thought I'd remove the "description" key which value is lengthy. WDYT?

Reply:
looks like you added the description as well, yes looks fine.